code/__HELPERS/matrices.dm

```diff
@@ -213,3 +213,12 @@ round(cos_inv_third+sqrt3_sin, 0.001), round(cos_inv_third-sqrt3_sin, 0.001), ro
 		else
 			CRASH("Invalid/unsupported color format argument in color_to_full_rgba_matrix()")
 
+/atom/proc/shake_animation(intensity = 8, time = 2) //Makes the object visibly shake
+	var/initial_transform = new/matrix(transform)
+	var/init_px = pixel_x
+	var/shake_dir = pick(-1, 1)
+	var/rotation = 2+soft_cap(intensity, 1, 1, 0.94)
+	var/offset = 1+soft_cap(intensity*0.3, 1, 1, 0.8)
+	var/time_new = time +soft_cap(intensity*0.3, 2, 1, 0.92)
```

Comment:
im not sure why youre using this soft cap function for time. if someone wants to set a specific time. they want to set a specific time.

Reply:
i removed the softcap in this animation, tested in game and it seems to be fine